spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -765,31 +767,13 @@ private DatastoreReaderWriter getDatastoreReadWriter() {
 		matcherAccessor.getPropertySpecifiers();
 		LinkedList<StructuredQuery.Filter> filters = new LinkedList<>();
 		persistentEntity.doWithColumnBackedProperties((persistentProperty) -> {
-
-			if (!example.getMatcher().isIgnoredPath(persistentProperty.getName())) {
-				// ID properties are not stored as regular fields in Datastore.
-				String fieldName = persistentProperty.getFieldName();
-				Value<?> value;
-				if (persistentProperty.isIdProperty()) {
-					PersistentPropertyAccessor accessor = persistentEntity.getPropertyAccessor(probe);
-					Object property = accessor.getProperty(persistentProperty);
-					value = property != null
-									? KeyValue.of(createKey(persistentEntity.kindName(), property))
-									: NullValue.of();
-				}
-				else {
-					value = probeEntity.getValue(fieldName);
-				}
-				if (value instanceof NullValue
-						&& example.getMatcher().getNullHandler() != ExampleMatcher.NullHandler.INCLUDE) {
-					//skip null value
-					return;
-				}
-				filters.add(StructuredQuery.PropertyFilter.eq(fieldName, value));
+			if (notIgnoredProperty(example, persistentProperty)) {
+				Value<?> value = getValue(example, probeEntity, persistentEntity, persistentProperty);
+				NullHandler nullHandler = example.getMatcher().getNullHandler();
```

Comment:
So the check for `value instanceof NullValue` is not needed?

Reply:
It is still required. NullHandler just indicates if we need to include null valued fields or not, and `value instanceof NullValue` is a check to see if a field holds a null value.

Reply:
Oh, it's in addFilter now. I missed it.